src/Apps/CollectorProfile/Routes/Saves2/Utils/prepareImageURLs.ts

```diff
@@ -0,0 +1,7 @@
+export const prepareImageURLs = (imageURLs: (string | null)[]) => {
+  // Ensure we have an array of exactly 4 images
+  return Array.from(Array(4)).reduce(
+    (acc, _, i) => [...acc, imageURLs[i] ?? null],
+    [] as string[]
+  )
+}
```

Comment:
**question**: should we add additional logic so that non-empty image URLs always go at the beginning of the array? Something like this `from ["url-1", null, "url-2", null] to ["url-1", "url-2", null, null]`

Reply:
What's the situation in which you'd be expecting to not have images? (I know it's possible but just should be rare enough to not worry too much about it)

Reply:
> I know it's possible but just should be rare enough to not worry too much about it

got it. I think then we don't have to worry about it (I was thinking of adding these changes as a safety net)